api/tests/mocha/controllers/login.spec.js

```diff
@@ -209,6 +209,16 @@ describe('login controller', () => {
       });
     });
 
+    it('matches the request header locale to the supported locales', () => {
+      req.headers = { 'accept-language': 'xx, fr' }; // xx as fake locale
+      const send = sinon.stub(res, 'send');
+      sinon.stub(fs, 'readFile').callsArgWith(2, null, 'LOGIN PAGE GOES HERE. {{ browserLocale }}');
+      sinon.stub(config, 'translate').returns('TRANSLATED VALUE.');
+
+      return controller.get(req, res).then(() => {
+        chai.expect(send.args[0][0]).to.equal('LOGIN PAGE GOES HERE. fr');
+      });
+    });
```

Comment:
It'd be good to add some more tests, eg:
1. to make sure that very specific locales match with our generic ones, for example, `en-NZ` should match with `en`.
2. what if none of the browser locales match?
3. what if the browser locale is undefined?

Reply:
Added!